File: scripts/rebuild.py
#!/usr/bin/env python3
import subprocess
import shutil
import os
from pathlib import Path

# 项目根目录（使用绝对路径）
# 获取脚本目录，然后往上一级是项目根
script_dir = Path("/vercel/share/v0-project/scripts")
root_dir = script_dir.parent

print(f"[v0] 脚本目录: {script_dir}")
print(f"[v0] 项目根目录: {root_dir}")
print(f"[v0] 开始清理 node_modules...")

# 清理 node_modules
node_modules = root_dir / "node_modules"
if node_modules.exists():
    print("[v0] 删除 node_modules...")
    shutil.rmtree(node_modules, ignore_errors=True)
    print("[v0] 已删除 node_modules")

# 运行 npm install
print(f"[v0] 在目录 {root_dir} 中运行 npm install...")
try:
    # 首先检查 package.json 是否存在
    pkg_json = root_dir / "package.json"
    if not pkg_json.exists():
        print(f"[v0] 错误: 找不到 package.json: {pkg_json}")
    else:
        print(f"[v0] 找到 package.json")
        result = subprocess.run(
            ["npm", "install"],
            cwd=str(root_dir),
            capture_output=False,
            text=True
        )
        if result.returncode == 0:
            print("[v0] 依赖安装成功！")
        else:
            print("[v0] npm install 返回非零代码")
except Exception as e:
    print(f"[v0] 出错: {e}")

File: scripts/rebuild.js
import { execSync } from "child_process";
import { rmSync, existsSync } from "fs";
import { join } from "path";

const rootDir = process.cwd();
const nodeModulesPath = join(rootDir, "node_modules");
const lockFilePath = join(rootDir, "package-lock.json");

console.log("[v0] 开始清理损坏的依赖...");

try {
  // 删除 node_modules
  if (existsSync(nodeModulesPath)) {
    console.log("[v0] 删除 node_modules 目录...");
    rmSync(nodeModulesPath, { recursive: true, force: true });
  }

  // 删除 package-lock.json
  if (existsSync(lockFilePath)) {
    console.log("[v0] 删除损坏的 package-lock.json...");
    rmSync(lockFilePath, { force: true });
  }

  // 运行 npm install
  console.log("[v0] 运行 npm install...");
  execSync("npm install", {
    cwd: rootDir,
    stdio: "inherit",
    env: { ...process.env, npm_config_legacy_peer_deps: "true" },
  });

  console.log("[v0] 依赖安装完成！");
} catch (error) {
  console.error("[v0] 安装过程中出错:", error.message);
  process.exit(1);
}
